Tests/BackendIntegrationTests/StoreKitIntegrationTests.swift

```diff
@@ -932,6 +935,51 @@ class StoreKit1IntegrationTests: BaseStoreKitIntegrationTests {
         self.assertNoActiveSubscription(customerInfo)
     }
 
+    func testVerifyPurchaseGrantsEntitlementsThroughOnRetryAfter429() async throws {
+
+        // Ensure that the first two times POST /receipt is called, we mock a 429 error
+        // and then proceed normally with the backend on subsequent requests
+        let host = try XCTUnwrap(HTTPRequest.Path.serverHostURL.host)
+        var requestCount = 0
+        stub(condition: isHost(host) && isPath("/v1/receipts")) { _ in
+            requestCount += 1
+
+            // Fail the first two requests, allow subsequent requests to go through to the backend
+            if requestCount < 3 {
+
+                if requestCount == 2 {
+                    HTTPStubs.removeAllStubs()
+                }
+                return Self.emptyTooManyRequestsResponse()
+            }
+
+            return .init()
+        }
+
+        let product = try await self.monthlyPackage.storeProduct
+        let customerInfo = try await self.purchases.purchase(product: product).customerInfo
+        try await self.verifyEntitlementWentThrough(customerInfo)
+        expect(customerInfo.allPurchasedProductIdentifiers) == [
+            product.productIdentifier
+        ]
+    }
+
+    func testVerifyPurchaseDoesntGrantEntitlementsAfter429RetriesExhausted() async throws {
```

Comment:
Is there a way we can test that exactly 3 retries were made here?

Reply:
Discussed this with Mark in Slack - we added in verifications for the number of times that the request was failed before eventually succeeding with the backend